### PR TITLE
my palindrome solution in go

### PR DESCRIPTION
**Pull request by drkennetz**
can be run in this directory by running `go test`. I've only handled the types of int and string in my type conversions (but I had a fun time writing a type converter). I also got my hands a little dirty with writing test tables!

### CODE REVIEW

**Review by birenderjit — COMMENTED** [review]
I don't think this will work for say AA11BB.
or we are not considering mixed letters and numbers.

File: challenges/2021-04-20-potential-palindrome/solutions/drkennetz.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	}
+}
+// CanMakePalindrome takes in any type, treats it as a string and checks to see if it can form a palindrome
+func CanMakePalindrome(x interface{}) bool {
+	s := convertToString(x)
+	if s == "" {
+		return false
+	}
+	// 256 chars in ascii charset
+	ords := make([]int, 256)
```
Comment (parhaml) [inline challenges/2021-04-20-potential-palindrome/solutions/drkennetz.go:36]: Fun idea to pre-make the map

File: challenges/2021-04-20-potential-palindrome/solutions/drkennetz.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	for pos := range(s) {
+		ords[s[pos]] = ords[s[pos]]+1
+	}
+	oddCount := 0
+	for i := range(ords) {
+		if ords[i]%2 != 0 {
+			oddCount++
+		}
+		if oddCount > 1 {
```
Comment (parhaml) [inline challenges/2021-04-20-potential-palindrome/solutions/drkennetz.go:45]: nice to interrupt as early as possible 💯

File: challenges/2021-04-20-potential-palindrome/solutions/drkennetz_test.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	}
+
+	for _, table := range tables {
+		result := CanMakePalindrome(table.x)
+		if result != table.y {
+			t.Error("FAIL: CanMakePalindrome of: ", result, "was incorrect, got: ", result, "expected: ", table.y)
+		} else {
+			fmt.Println("SUCCESS: CanMakePalindrome of: ", result, "was correct, got: ", result, "expected: ", table.y)
+		}
+	}
```
Comment (parhaml) [inline challenges/2021-04-20-potential-palindrome/solutions/drkennetz_test.go:26]: really cool

**Review by parhaml — APPROVED** [review]

**Comment by drkennetz** [comment]
> I don't think this will work for say AA11BB.
> or we are not considering mixed letters and numbers.

If they are converted to string then ranging over them should return each character’s ascii value. Numbers have an ascii encoding also in the 256 charset so it should work. I am incrementing the index in the int slice of 0s that corresponds to the ascii encoding. I’ll add it to the test table tomorrow to make sure, but I don’t see why it wouldn’t work.

**Review by ericwburden — APPROVED** [review]

**Comment by drkennetz** [comment]
@birenderjit just to follow up, I added that test case to the table and it passed.

**Comment by birenderjit** [comment]
Nice!!

**Pull request merged**